<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>dscannercheck::ProjectConfigPage</class>
 <widget class="QWidget" name="dscannercheck::ProjectConfigPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>443</width>
    <height>502</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabChecks">
      <attribute name="title">
       <string comment="@title:tab">Checks</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="0">
          <widget class="QCheckBox" name="kcfg_checkUnusedFunction">
           <property name="toolTip">
            <string comment="@info:tooltip">Check for unused functions. It is recommend to only enable this when the whole program is scanned</string>
           </property>
           <property name="text">
            <string comment="@option:check">Unused function</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="kcfg_inconclusiveAnalysis">
           <property name="toolTip">
            <string comment="@info:tooltip">Allow that Cppcheck reports even though the analysis is inconclusive. There are false positives with this option. Each result must be carefully investigated before you know if it is good or bad.</string>
           </property>
           <property name="text">
            <string comment="@option:check">Inconclusive analysis</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="kcfg_checkInformation">
           <property name="toolTip">
            <string comment="@info:tooltip">Enable information messages</string>
           </property>
           <property name="text">
            <string comment="@option:check">Information</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="kcfg_checkMissingInclude">
           <property name="toolTip">
            <string comment="@info:tooltip">Warn if there are missing includes. For detailed information use 'check-config'</string>
           </property>
           <property name="text">
            <string comment="@option:check">Missing include</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="kcfg_checkPerformance">
           <property name="toolTip">
            <string comment="@info:tooltip">Enable performance messages</string>
           </property>
           <property name="text">
            <string comment="@option:check">Performance</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="kcfg_forceCheck">
           <property name="toolTip">
            <string comment="@info:tooltip">&lt;p&gt;Force checking of files that have a lot of configurations. Error is printed if such a file is found so there is no reason to use this by default. If used together with '--max-configs=', the last option is the one that is effective.&lt;br/&gt;&lt;br/&gt;Be careful with using this option - it can dramatically slow checking.&lt;/p&gt;</string>
           </property>
           <property name="text">
            <string comment="@option:check">Force checking</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="kcfg_checkPortability">
           <property name="toolTip">
            <string comment="@info:tooltip">Enable portability messages</string>
           </property>
           <property name="text">
            <string comment="@option:check">Portability</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="kcfg_checkStyle">
           <property name="toolTip">
            <string comment="@info:tooltip">Enable all coding style checks. All messages with the severities 'style', 'performance' and 'portability' are enabled.</string>
           </property>
           <property name="text">
            <string comment="@option:check">Style</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QCheckBox" name="kcfg_checkConfig">
           <property name="toolTip">
            <string comment="@info:tooltip">Check Cppcheck configuration. The normal code analysis is disabled by this flag.</string>
           </property>
           <property name="text">
            <string comment="@option:check">Check config</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabIncludes">
      <attribute name="title">
       <string comment="@title:tab">Include Directories</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QCheckBox" name="kcfg_useProjectIncludes">
         <property name="toolTip">
          <string comment="@info:tooltip">Add 'project' include directories to Cppcheck arguments. This option adds only paths related with project's root and build directories.</string>
         </property>
         <property name="text">
          <string comment="@option:check">Use 'project' include dirs</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="kcfg_useSystemIncludes">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string comment="@info:tooltip">&lt;p&gt;Add other, 'system' include directories to Cppcheck arguments.&lt;br/&gt;&lt;br/&gt;Be careful with using this option - it can dramatically slow checking.&lt;/p&gt;</string>
         </property>
         <property name="text">
          <string comment="@option:check">Use 'system' include dirs</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string comment="@label:textbox">&amp;Ignored includes:</string>
           </property>
           <property name="buddy">
            <cstring>kcfg_ignoredIncludes</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="kcfg_ignoredIncludes">
           <property name="toolTip">
            <string comment="@info:tooltip">&lt;html&gt;

&lt;p&gt;Defines semicolon-separated list of ignored include directories.&lt;/p&gt;

&lt;p&gt;You can use the following placeholders:&lt;/p&gt;

&lt;p&gt;&lt;b&gt;&lt;tt&gt;%p&lt;/tt&gt;&lt;/b&gt; - Gets replaced by the URL of the project's root directory.&lt;/p&gt;

&lt;p&gt;&lt;b&gt;&lt;tt&gt;%b&lt;/tt&gt;&lt;/b&gt; - Gets replaced by the URL of the project's build directory.&lt;/p&gt;

&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabExtra">
      <attribute name="title">
       <string comment="@title:tab">Extra Parameters</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="extraParametersLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <item>
            <widget class="QLabel" name="extraLabel">
             <property name="text">
              <string comment="@label:textbox">Ex&amp;tra parameters:</string>
             </property>
             <property name="buddy">
              <cstring>kcfg_extraParameters</cstring>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="kcfg_extraParameters">
             <property name="toolTip">
              <string comment="@info:tooltip">&lt;html&gt;

&lt;p&gt;Defines additional parameters for cppcheck (see documentation).&lt;/p&gt;

&lt;p&gt;You can use the following placeholders:&lt;/p&gt;

&lt;p&gt;&lt;b&gt;&lt;tt&gt;%p&lt;/tt&gt;&lt;/b&gt; - Gets replaced by the URL of the project's root directory.&lt;/p&gt;

&lt;p&gt;&lt;b&gt;&lt;tt&gt;%b&lt;/tt&gt;&lt;/b&gt; - Gets replaced by the URL of the project's build directory.&lt;/p&gt;

&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string comment="@title:group">Command Line</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLineEdit" name="commandLineFilter">
          <property name="placeholderText">
           <string comment="@info:placeholder">Search...</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="commandLineBreaks">
          <property name="text">
           <string comment="@option:check">Break lines</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="commandLine">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="KMessageWidget" name="messageWidget">
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="closeButtonVisible">
      <bool>false</bool>
     </property>
     <property name="messageType">
      <enum>KMessageWidget::Error</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header>KMessageWidget</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>kcfg_checkStyle</tabstop>
  <tabstop>kcfg_checkPerformance</tabstop>
  <tabstop>kcfg_checkPortability</tabstop>
  <tabstop>kcfg_checkInformation</tabstop>
  <tabstop>kcfg_checkUnusedFunction</tabstop>
  <tabstop>kcfg_checkMissingInclude</tabstop>
  <tabstop>kcfg_inconclusiveAnalysis</tabstop>
  <tabstop>kcfg_forceCheck</tabstop>
  <tabstop>kcfg_checkConfig</tabstop>
  <tabstop>kcfg_useProjectIncludes</tabstop>
  <tabstop>kcfg_useSystemIncludes</tabstop>
  <tabstop>kcfg_ignoredIncludes</tabstop>
  <tabstop>kcfg_extraParameters</tabstop>
  <tabstop>commandLineBreaks</tabstop>
  <tabstop>commandLine</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>kcfg_useProjectIncludes</sender>
   <signal>toggled(bool)</signal>
   <receiver>kcfg_useSystemIncludes</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>665</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
